### ISSUE TITLE
0.1 release

### ISSUE THREAD

**Issue by MechCoder** [scikit-optimize/scikit-optimize#79]
I would like to get the 0.1 release out before school starts again (i.e September). This is just a parent issue to track the blockers.

- [x]  Consistent and backward-compatible API. Addressed by https://github.com/scikit-optimize/scikit-optimize/pull/75
- [x] SMAC https://github.com/scikit-optimize/scikit-optimize/issues/57
- (Local search technique that performs better than random sampling on piecewise constant predict functions (https://github.com/scikit-optimize/scikit-optimize/issues/74), postponed till we have a conclusion in #109)
- [x] Examples (https://github.com/scikit-optimize/scikit-optimize/issues/65)
- [x] Support for Python 2.7 (#87)
- [x] Consistent return types #86 
- [x] Name collision #76 ([punting for now](https://github.com/scikit-optimize/scikit-optimize/issues/76#issuecomment-235634085))
- [x] Need a logo #107 (code speaks louder than images, no logo required)
- [ ] release mechanics #133
- [x] better defaults #166 
- [x] merge #145 
- [x] merge #169
- [x] maybe merge #162 (nice to have but don't hold the 🚄 for it)
- [x] stop this list from getting ever longer 📋

Is there anything else?

**Comment by glouppe** [comment]
Adding #87 as part of the release. Should be easy.

**Comment by glouppe** [comment]
For the examples, which ones do you think we do for the release? #102? #103? both? more?

**Comment by glouppe** [comment]
Also, do you want to have full support of categorical variables in `gp_minimize` for that release? (I would postpone it)

**Comment by betatim** [comment]
I think there will be more people who would be interested in the use-case shown in #102.

What do we do about mixed types (#86) for 0.1?

**Comment by glouppe** [comment]
> What do we do about mixed types (#86) for 0.1?

Yes, this should be fixed! I am adding that to the list.

**Comment by glouppe** [comment]
Only two items left!!

**Comment by betatim** [comment]
two, one, ... 🚀

**Comment by betatim** [comment]
Should we wait on #76 being resolved or do we think that is something that will take too long? We have some good momentum going right now which I think we should keep going.

**Comment by MechCoder** [comment]
I have a smarter idea to optimize the acquisition function, hope to try it out after work today. ::D

**Comment by MechCoder** [comment]
I meant that as a general comment and not for the release. Yes I think we should resolve the name-clash.

**Comment by MechCoder** [comment]
> Also, do you want to have full support of categorical variables in gp_minimize for that release? 

I will postpone that as well.

**Comment by betatim** [comment]
Updated the to do list. Basically punting #76 and #107 to the future.

The conclusions at the end of #109 are inconclusive so shall we postpone that as well? I vote yes.

**Comment by glouppe** [comment]
> The conclusions at the end of #109 are inconclusive so shall we postpone that as well? I vote yes.

+1

**Comment by betatim** [comment]
(edited checklist)

**Comment by MechCoder** [comment]
edited again.

**Comment by glouppe** [comment]
Once #174 is merged, I'd like to go ahead and make the release.  What do you guys think?

**Comment by betatim** [comment]
If you are happy with #133 (aka would use it or add what is missing) then yes lets release! I will be out most of the day doing not-near-a-computer things but you don't really need me anyway :)

**Comment by betatim** [comment]
We should add the wonderful plotting to `CHANGELOG.md`.

**Comment by glouppe** [comment]
> We should add the wonderful plotting to CHANGELOG.md.
I did it as part of #175  ;) But I'll add one of those nice plots as part of the index.

**Comment by glouppe** [comment]
> If you are happy with #133 (aka would use it or add what is missing) then yes lets release! I will be out most of the day doing not-near-a-computer things but you don't really need me anyway :)

Great! I'll wait for Manoj's opinion and then go ahead.

**Comment by MechCoder** [comment]
Can we just merge https://github.com/scikit-optimize/scikit-optimize/pull/145 and https://github.com/scikit-optimize/scikit-optimize/issues/162 ? (Both should not take much time to review)

@betatim Would you like to add https://github.com/scikit-optimize/scikit-optimize/pull/169 before the release?

Also about narrative documentation, are we pushing that to after the release (I'm +1 for moving that to after the release as well)?

**Comment by glouppe** [comment]
> Can we just merge #145 and #162 ? (Both should not take much time to review)

Sure no problem :)

**Comment by betatim** [comment]
For #162 I am Ok with doing that post 0.1. Mainly because I can't promise to have time to work on it till the start of next week.

**Comment by betatim** [comment]
For #169: we can do 0.1 without I think. I see two options for the example:

1. really dig into the differences between GP and forest. Needs a lot of work from our side to understand what we are saying.
2. make it as brief as possible and show off how easy it is to swap between GP and ET

1. would be "better" but also a lot of work, so maybe we go for 2. to get v0.1 to converge and then expand the example a la option 1. post-0.1.

Same comment as before applies: I am away this week so hard for me to commit any time to reviewing things before next week.

**Comment by MechCoder** [comment]
I really think we should understand what we are saying. I'll update the notebook with my thoughts today evening.

> Same comment as before applies: I am away this week so hard for me to commit any time to reviewing things before next week.

That is fine; No rush. We still are a month behind the initial deadline. Delaying by 2-3 days would not matter much.

**Comment by MechCoder** [comment]
Oh sorry, I meant ahead of the deadline or whatever the correct phrase is. (Pardon my not-so-great English :p )

**Comment by betatim** [comment]
Added #145 and #169 to the list.

**Comment by betatim** [comment]
This means we have to make a release now right? Or can we find another excuse to not do it :trollface:

If not, I will make a release in an hour or so. Yay or nay?

**Comment by glouppe** [comment]
> Yay or nay?

Yay!

**Comment by glouppe** [comment]
Closing!

**Comment by betatim** [comment]
🍻 

Thanks everyone!

(need to level up my python packaging and dependencies skill)

**Comment by raghavrv** [comment]
:beers:
